native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java

```diff
@@ -111,14 +109,30 @@ public static SyntaxTree generateSyntaxTree(StubFile stubFile, boolean isRoot) {
                     "ballerina", "grpc"
             );
             imports = imports.add(importForGrpc);
-            if (checkForImportsInStub(stubFile, GOOGLE_PROTOBUF_TIMESTAMP_PROTO)
-                    || checkForImportsInStub(stubFile, GOOGLE_PROTOBUF_DURATION_PROTO)) {
-                ImportDeclarationNode importForTime = Imports.getImportDeclarationNode(
-                        "ballerina", "time"
-                );
-                imports = imports.add(importForTime);
+        }
+        boolean importTime = false;
+        for (ServiceStub serviceStub : stubFile.getStubList()) {
+            List<Method> methodList = new ArrayList<>();
+            methodList.addAll(serviceStub.getUnaryFunctions());
+            methodList.addAll(serviceStub.getClientStreamingFunctions());
+            methodList.addAll(serviceStub.getServerStreamingFunctions());
+            methodList.addAll(serviceStub.getBidiStreamingFunctions());
+
+            if (checkForImportsInServices(methodList, "time:Utc")
```

Comment:
```suggestion
            if (!importTime && (checkForImportsInServices(methodList, "time:Utc")
```
Maybe we can avoid redundant checks in this way

Reply:
Added a break in 5b8a515